cachi2/core/package_managers/yarn/resolver.py

```diff
@@ -0,0 +1,197 @@
+"""
+Resolve the dependency list for a yarn project.
+
+It also performs the necessary validations to avoid allowing an invalid
+project to keep being processed.
+"""
+import json
+import logging
+import re
+from typing import NamedTuple, TypedDict
+
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.yarn.utils import run_yarn_cmd
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+
+_PACKAGE_PROTOCOL_REGEX = re.compile("@[a-z]+:")
+
+
+class Dependency(TypedDict):
+    """A dependency listed in a yarn Package.
+
+    See 'yarn info' output. Example:
+
+    {
+        "descriptor": "yargs@npm:^17.7.2",
+        "locator": "yargs@npm:17.7.2"
+    }
+    """
+
+    descriptor: str
+    locator: str
+
+
+class Package(NamedTuple):
+    """A package listed by the yarn info command.
+
+    See the 'yarn info' output. Example:
+
+    {
+        "value": "@types/yargs@npm:17.0.24",
+        "children": {
+            "Version": "17.0.24",
+            "Dependencies": []
+        }
+    }
+
+    Note that the attributes present in the json vary a little depending on the
+    type of dependency.
+    """
+
+    locator: str
+    version: str
+    dependencies: list[Dependency]
```

Comment:
I don't really think we will have a use for the `dependencies` unless we decide to implement dev-dependency detection

Reply:
Removed.